sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -234,6 +315,8 @@ __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
     return tmp;
   }
 
+  reference operator->() const { return reference(m_Ptr); }
```

Comment:
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc#representation-of-annotated_ptr section should be updated.